src/v/cloud_topics/frontend/frontend.cc

```diff
@@ -548,13 +548,30 @@ ss::future<> bg_upload_and_replicate(
         op->replicate_finished.set_value(raft::errc::timeout);
     });
 
+    /*
+     * L0 GC relies on a minimum epoch associated with each NTP for calculating
+     * the name of an L0 object. The minimum is based on the topic revision, but
+     * from the perspective of cloud topics this is largely an implementation
+     * detail. So here we go ahead and translate into a cluster_epoch type for
+     * the rest of its journey.
+     */
+    auto min_epoch = cluster_epoch(partition->get_topic_revision_id());
```

Comment:
I wonder to keep all these things in one place if our cluster epoch service should do this conversion.